src/constructor/constructor.py

```diff
@@ -0,0 +1,259 @@
+        """
+        if phase not in ['train', 'valid', 'test', 'predict']:
+            raise ValueError(f'Not support phase for data loaders specification: {phase}')
+
+        return [
+            self.__prepare_dataloader(phase_params.dataset, phase_params.dataloader)
+            for phase_params in self.hparams.data['phase']
+        ]
+
+    @staticmethod
+    def __prepare_dataloader(dataset_params: DictConfig, dataloader_params: DictConfig) -> DataLoader:
+        dataset = Constructor.__create_dataset(dataset_params)
+        collate_fn = dataset.collate_fn if hasattr(dataset, 'collate_fn') else None
+
+        loader = DataLoader(dataset=dataset,
+                            collate_fn=collate_fn,
+                            **dataloader_params)
+
+        return loader
+
+    @staticmethod
+    def __create_dataset(dataset_params: DictConfig) -> ImageDataset:
+        transform = Constructor.__create_transforms(dataset_params.transform)
+        augment = Constructor.__create_transforms(dataset_params.augment)
+
+        dataset_class = DATASETS.get(dataset_params.name)
+
+        return dataset_class(transform=transform, augment=augment, **dataset_params.params)
+
+    @staticmethod
+    def __prepare_transforms_recursively(transforms: ListConfig[DictConfig]) -> List[A.Compose, A.BaseCompose]:
+        transforms_list = []
+
+        for transform_info in transforms:
+            transform_name = transform_info.name
+            transform_params = transform_info.params
+
+            if transform_name == 'Compose' or transform_name == 'OneOf' or transform_name == 'SomeOf' or \
+                    transform_name == 'PerChannel' or transform_name == 'Sequential':
+                transform = Constructor.__prepare_base_compose(transform_name, **transform_params)
+            elif transform_name == 'OneOrOther':
+                raise ValueError('OneOrOther composition is currently not supported')
+            else:
+                transform = TRANSFORMS.get(transform_name)(**transform_params)
+
+            transforms_list.append(transform)
+
+        return transforms_list
+
+    @staticmethod
+    def __prepare_base_compose(compose_name: str, **kwargs) -> A.Compose:
+        transforms = kwargs.pop('transforms', None)
+        if transforms is None:
+            raise ValueError(f'There are transforms must be specified for {compose_name} composition')
+
+        transforms_list = Constructor.__prepare_transforms_recursively(transforms)
+        transform = TRANSFORMS.get(compose_name)(transforms=transforms_list, **kwargs)
+
+        return transform
+
+    @staticmethod
+    def __create_transforms(transforms_params: ListConfig[DictConfig]) -> Optional[A.Compose]:
+        if transforms_params is None:
+            return None
+
+        return Constructor.__prepare_base_compose('Compose', transforms=transforms_params)
+
+    def configure_metrics_manager(self):
```

Comment:
It would be:
`def configure_metrics_manager(self) -> MetricManager:
        """Create list of metrics wrapping them into a MetricManager module.
        
        Returns: MetricManager module.
        """
        metric_manager = MetricManager(self.__hparams.metrics)
        return metric_manager`